text/0155-external-modules.md

```diff
@@ -0,0 +1,124 @@
+We hope that wider community will integrate their projects with OpenTelemetry.
+We have to encourage that by providing great documentation, examples and tooling to integration authors,
+while still providing our end-users with some way to discover all available Otel components together with some visibility into their quality.
+
+## Explanation
+
+The [OpenTelemetry Registry](https://opentelemetry.io/registry/) serves as a central catalogue of all known Otel components,
+both provided by core maintainers of the project and any third party.
+
+In order for a component to be included into Registry its authors have to fill a [self-assessment form](#registry-self-assessment-form).
+
+The registry should allow a clear visibility of component's ownership, quality and compatibility with other Otel components.
+
+A component can be removed from the Registry if any declaration from the self-assessment form is violated and not remedied
+in a timely manner, provisionally one or two weeks.
+
+## Internal details
+
+We distinguish the following sources of OpenTelemetry components and integrations.
+
+### Native or built-in instrumentations
+
+Any library or framework can use the OpenTelemetry API to natively produce telemetry.
+We encourage library authors to submit their library for inclusion into the OpenTelemetry Registry.
+
+### Core components
+
+Otel SIGs may provide instrumentation for any libraries that are deemed important enough by the SIG’s maintainers.
+By doing this the SIG maintainers commit to support this instrumentation (including future versions of the library),
+provide updates/fixes, including security patches and guarantee that they comply with Otel semantic conventions and best practices.
+
+Depending on the SIG, these instances of core instrumentation may share the repository, infrastructure and maintainers with
+the OpenTelemetry API/SDK implementation for this language or be separate.
+
+All instances of core instrumentation must be included into the Otel Registry following the usual process described above.
+
+### Contrib components
+
+Any language SIG may have one or more “contrib” repos containing components contributed by external developers.
+These repositories may have a separate set of approvers/maintainers than the core API/SDK repo.
+SIG maintainers are encouraged to have a simplified process to promote contributors to approver/maintainer role in this repository.
+A contrib repository may leverage the CODEOWNERS functionality of GitHub to assign maintainers to individual packages
+even if this means granting write permissions to the whole repo.
+The goal should be to distribute the load of reviewing PRs and accepting changes as much as possible.
+
+All components in a contrib repository are expected to be included into the Otel Registry following the usual process described above.
+
+It should be very easy for external contributors to include their components into a contrib repo as opposed to hosting them separately.
```

Comment:
> as opposed to hosting them separately

What do we expect to simplify for contributors? Security scanning and testing below is not a necessity (it is for us, but may not be for contributor). I think the intent of this sentence is different from how it may be interpreted. It may be harder as we likely will ask for things like build definitions, proper readme files, etc. which may not be the same in separate externally owned repo.

Maybe rephrase it emphasizing that maintainers of contrib repo should be open for adding new components

Reply:
@SergeyKanzhelev I tried to rephrase this, PTAL.